src/XMakeTasks/Microsoft.CSharp.CurrentVersion.targets

```diff
@@ -46,6 +46,18 @@ Copyright (C) Microsoft Corporation. All rights reserved.
         <SynthesizeLinkMetadata Condition=" '$(SynthesizeLinkMetadata)' == '' and '$(HasSharedItems)' == 'true' ">true</SynthesizeLinkMetadata>
     </PropertyGroup>
 
+    <PropertyGroup>
+        <!-- Select a default C# compiler here, if unspecified, instead of in the Csc task, so that we can also
+             decide on the debug extension to be used. -->
+        <CscToolExe Condition="'$(CscToolExe)' == '' and '$(MSBuildRuntimeType)' != 'Mono'">csc.exe</CscToolExe>
+        <CscToolExe Condition="'$(CscToolExe)' == '' and '$(MSBuildRuntimeType)' == 'Mono'">mcs.exe</CscToolExe>
```

Comment:
It should be, but `mcs` is what is installed by default currently, so, this is a good default when running on Mono. Otherwise anyone trying to use msbuild with mono and not having csc.exe will have to set `CscToolExe` on the command line. We could set this in a `msbuild` wrapper script, but then it overrides any value specified in the targets files, like for Xamarin Android/iOS. Also, we need to know the compiler being used to figure out the debug file extension to use.

The reasoning is explained in the commit message too.